33/issuenumber5/calendarsettings

ConstantinLange opened the pull request:
You can enter a startdate and an enddate for every event, as well as a duration. The validations check, that an enddate can not be before a startdate and the duration corresponds with the dates.

ConstantinLange committed:
Added startedate for event with corresponding tests.  #33

ConstantinLange committed:
Added enddate and duration for an event with corresponding events. #33

ConstantinLange committed:
Added validations for start- and enddate and for the duration. #33

sumpfranze commented on spec/models/event_spec.rb:
```diff
@@ -1,14 +1,52 @@
+  it "should have a deadline, startdate, enddate and duration" do
+    @event = Event.create(name: "NAME", description: "DESCRIPTION", gamemode: "GAMEMODE",
+    sport: "SPORT", teamsport: false, playercount: 1, gamesystem: "GAMESYSTEM",
+    deadline: nil, startdate: nil, enddate: nil, duration: nil)
```
you can use the FactoryBot for that and just set the desired values to nil.

sumpfranze replied:
those four attributes should be tested for nil separately, since each validation for the attribute must be tested separately. This test would also be green, even if there is just a validation for one of the values.

Also it would give you more information on what exactly fails when you run the test.

sumpfranze commented on spec/models/event_spec.rb:
```diff
@@ -1,14 +1,52 @@
+    @event = FactoryBot.create :event
+    @date = Date.new(2017,11,16)
+    expect(@event.deadline).to eq(@date)
+  end
```
aren't "should have attribute" tests the tests for validation? So should also check, that the attribute can't be nil.

sumpfranze commented on spec/models/event_spec.rb:
```diff
@@ -1,14 +1,52 @@
+    sport: "SPORT", teamsport: false, playercount: 1, gamesystem: "GAMESYSTEM",
+    deadline: Date.new(2017,11,16), startdate: Date.new(2017,12,01), enddate: Date.new(2017,12,05), duration: 6)
+
+    expect(@event).not_to be_valid
```
you also need to test the positive. So create an event with a startdate, enddate and duration where the values are correct and test whether that is valid or not.

sumpfranze commented on spec/models/event_spec.rb:
```diff
@@ -1,14 +1,52 @@
+    sport: "SPORT", teamsport: false, playercount: 1, gamesystem: "GAMESYSTEM",
+    deadline: Date.new(2017,11,16), startdate: Date.new(2017,12,05), enddate: Date.new(2017,12,01), duration: 10)
+  
+    expect(@event).not_to be_valid
```
It should be possible the other way around, so that there can be a valid event, when startdate < enddate.

sumpfranze reviewed (changes requested):
A general question, do we even need to save the duration in the database? Wouldn't it make way more sense, if the duration is just a value that is calculated in a function in the model? Would save a bit of headache with validations and reduce redundancy.

ConstantinLange committed:
Added JavaScript to enter date easier. Ref.: #33

ConstantinLange committed:
Removed all IDE files. Ref.: #33

ConstantinLange committed:
Merge branch 'dev' into 33/issuenumber5/calendarsettings

sumpfranze committed:
Fix travis build
, fix code climate partially, show startdate, enddate, deadline on index page. Ref #33.

sumpfranze committed:
code climate. Ref #33.

sumpfranze committed:
Delete comment. Ref #33.

sumpfranze committed:
Maybe fix codeclimate now? Ref #33.
